internal/xds/translator/extension.go

```diff
@@ -81,9 +81,7 @@ func processExtensionPostVHostHook(vHost *routev3.VirtualHost, em *extensionType
 
 	// If the extension returned a modified Virtual Host, then copy its to the one that was passed in as a reference
 	if modifiedVH != nil {
-		// Overwrite the pointer for the original virtual host.
-		// Uses nolint because of the ineffectual assignment check
-		vHost = modifiedVH //nolint
+		copyPtr(modifiedVH, vHost)
```

Comment:
can we use https://pkg.go.dev/github.com/golang/protobuf/proto#Clone here instead ?

Reply:
I used it to get rid of the duplicated function in the `testutils` functions since those setup new variables, but it won't work for this file since it returns a value, so you end up in the same scenario I described where `*x = *y` and `*x = *proto.Clone(y)` cause lint errors because of the mutex inside all the xDS resource messages, and doing `x = proto.Clone(y)` results in x not really getting updated outside of the function since it's a local variable.

I was looking at `proto.Merge` as a potential solution, but unfortunately that won't work either since if you don't `.Reset()` the initial `Route/VirtualHost` first, then `Merge()` results in all the fields that are lists (like add request headers) getting duplicate entries. If you do call `.Reset()` on the `Route/VirtualHost` first, then `proto.Merge()` does nothing since it only merges fields that are set on the source and dest.